liesel/model/nodes.py

```diff
@@ -1100,6 +1103,159 @@ def all_input_vars(self) -> tuple[Var, ...]:
+            original_dist = original_var.dist_node.distribution
+            dist_inputs = original_var.dist_node.inputs
+
+            # transform the distribution
+            new_dist = tfd.TransformedDistribution(original_dist, tfb.Invert(bijector))
+
+            # transform initial value
+            new_value = bijector.inverse(original_var.value)
+
+            # initialise the new variable
+            new_var = lsl.Var(
+                new_value,
+                lsl.Dist(new_dist, *dist_inputs),
+                name=f"{original_var.name}_transformed"
+            )
+            new_var.parameter = original_var.parameter
+
+            # define the original variable as a function of the new variable
+            original_var.value_node = lsl.Calc(bijector.forward, new_var)
+            original_var.parameter = False
+
+            # return the new variable
+            return new_var
+
+        The value of the attribute :attr:`~liesel.model.nodes.Var.parameter` is
+        transferred to the transformed variable and set to ``False`` on the original
+        variable. The attributes :attr:`~liesel.model.nodes.Var.observed` and
+        :attr:`~liesel.model.nodes.Var.role` have the default values for
+        the transformed variable and remain unchanged on the original variable.
+
+        Examples
+        --------
+
+        >>> import tensorflow_probability.substrates.jax.distributions as tfd
+        >>> import tensorflow_probability.substrates.jax.bijectors as tfb
+
+        Assume we have a variable ``scale`` that is constrained to be positive, and
+        we want to include the log-transformation of this variable in the model.
+        We first set up the parameter var with its distribution:
+
+        >>> prior = lsl.Dist(tfd.HalfCauchy, loc=0.0, scale=25.0)
+        >>> scale = lsl.param(1.0, prior, name="scale")
+
+        The we transform the variable to the log-scale:
+
+        >>> log_scale = scale.transform(tfb.Exp())
+        >>> log_scale
+        Var(name="scale_transformed")
+
+        Now the ``log_scale`` has a log probability, and the ``scale`` variable does
+        not:
+
+        >>> log_scale.update().log_prob
+        Array(-3.6720574, dtype=float32)
+
+        >>> scale.update().log_prob
+        0.0
+        """
+        if self.weak:
+            raise RuntimeError(f"{repr(self)} is weak")
+
+        if self.dist_node is None:  # type: ignore
+            raise RuntimeError(f"{repr(self)} has no distribution")
+
+        if isinstance(bijector, type) and not bijector_args and not bijector_kwargs:
+            raise ValueError(
+                "You passed a bijector class instead of an instance, but did not "
+                "provide any arguments for the bijector. You should either provide "
+                "arguments or pass an instance of the bijector class instead."
+            )
+
+        # avoid infinite recursion
+        self.auto_transform = False
+
```

Comment:
could handle the `bijector is None` case above line 1233 and remove duplicated code:
```
if bijector is None:
            dist_inst = self.dist_node.init_dist()
            bijector = dist_inst.experimental_default_event_space_bijector
            
if isinstance(bijector, type):
            tvar = _transform_var_with_bijector_class(
                self, bijector, *bijector_args, **bijector_kwargs
            )
else: #bijector is bijector instance
   ...
```

Reply:
See my comment below, this  suggestion alerted me to a problem in my implementation. Thanks!